Improve ML how-to guides (#3602)
viamrobotics/docs: docs/how-tos/detect-people.md

---
title: "Detect objects in a camera stream"
linkTitle: "Detect objects or people"
type: "docs"
tags: ["vision", "data", "services", "quickstart", "ml", "camera", "webcam"]
no_list: true

> description: "Detect people on a camera stream from any webcam using the vision service."

Author: ```suggestion
description: "Detect people and their location in an image with any webcam and a vision service."
```

platformarea: ["ml"]
level: "Beginner"
date: "2024-07-31"
updated: "2024-10-20"
cost: "0"
---

WebRTC is a powerful technology that allows developers to build apps with video streams.
Like an app that provides you a live stream of your house's security cameras.

Adding Computer Vision allows machines to analyze images and gain meaningful information from video streams.
You can then program the machines to act based on this data, for example by alerting you when people appear on your camera stream.

> In this guide you'll use machine learning to detect people in a camera stream.

Author: ```suggestion
In this guide you'll use a publicly available machine learning model to detect people on a camera stream.
```

Click the **+** icon next to your machine part in the left-hand menu and select **Component**.
Select the `camera` type, then select the `webcam` model.
Enter a name or use the suggested name for your camera and click **Create**.

Click the **Save** button in the top right corner of the page to save your configuration.
Then click on the **Test** panel at the bottom of the camera's configuration panel to test the camera stream.

If you don't see an image stream, use the video path dropdown to select your camera path.

For more detailed configuration information and troubleshooting, see the [`webcam` docs](/components/camera/webcam/).

{{% /expand%}}
{{%expand "Step 4: Deploy a person detection model" %}}

In this guide you'll use a publicly available model to detect people.
You will deploy this model to your machine using the ML model service.

Click **+**, click **Service** and select the `ML model` type, then select the `TFLite CPU` model.
Create the service.

In the resulting ML model service configuration pane, ensure that **Deploy model on machine** is selected for the **Deployment** field.

> THen click on **Select model**, switch to the **Registry** tab and select the **people** model by **ml-models-scuttle** to deploy a model that has been trained to be able to detect people.

Author: ```suggestion
Then click on **Select model**, switch to the **Registry** tab and select the **people** model by **ml-models-scuttle** to deploy a model that has been trained to be able to detect people.
```

Click the **Save** button in the top right corner of the page to save your configuration and close and reopen the **Test** panel of the vision service configuration panel.
Now if you view detections, you will only see detections with a confidence value higher than the `"default_minimum_confidence"` attribute.

{{<imgproc src="/get-started/quickstarts/vision-card.png" resize="x1100" declaredimensions=true alt="Positive person detection on the vision card." >}}

{{% /expand%}}

## Next steps

You can now detect people or other objects on a camera stream using any device and any webcam.

> You can also use the Viam platform to train your own Machine Learning models or to configure alerts for when something is detected:

Author: ```suggestion
If you need a model to detect objects specific to your use case, you can use the Viam platform to train your own models or upload externally-trained models:
```
